src/servers/src/elasticsearch.rs

```diff
@@ -0,0 +1,366 @@
+    let log_values = match convert_es_input_to_log_values(&payload, &params.msg_field) {
+        Ok(log_values) => log_values,
+        Err(e) => {
+            return (
+                StatusCode::BAD_REQUEST,
+                elasticsearch_headers(),
+                axum::Json(write_bulk_response(
+                    start.elapsed().as_millis() as i64,
+                    0,
+                    StatusCode::BAD_REQUEST.as_u16() as u32,
+                    e.to_string().as_str(),
+                )),
+            );
+        }
+    };
+    let log_num = log_values.len();
+
+    if let Err(e) = ingest_logs_inner(
+        log_state.log_handler,
+        pipeline,
+        None,
+        table,
+        log_values,
+        Arc::new(query_ctx),
+    )
+    .await
+    {
+        warn!("Failed to ingest logs: {:?}", e);
+        return (
+            status_code_to_http_status(&e.status_code()),
+            elasticsearch_headers(),
+            axum::Json(write_bulk_response(
+                start.elapsed().as_millis() as i64,
+                0,
+                e.status_code() as u32,
+                e.to_string().as_str(),
+            )),
+        );
+    }
+
+    (
+        StatusCode::OK,
+        elasticsearch_headers(),
+        axum::Json(write_bulk_response(
+            start.elapsed().as_millis() as i64,
+            log_num,
+            StatusCode::CREATED.as_u16() as u32,
+            "",
+        )),
+    )
+}
+
+// It will generate the following response when write _bulk request to GreptimeDB successfully:
+// {
+//     "took": 1000,
+//     "errors": false,
+//     "items": [
+//         { "create": { "status": 201 } },
+//         { "create": { "status": 201 } },
+//         ...
+//     ]
+// }
+// If the status code is not 201, it will generate the following response:
+// {
+//     "took": 1000,
+//     "errors": true,
+//     "items": [
+//         { "create": { "status": 400, "error": { "type": "illegal_argument_exception", "reason": "<error_reason>" } } }
+//     ]
+// }
+fn write_bulk_response(took_ms: i64, n: usize, status_code: u32, error_reason: &str) -> Value {
+    if error_reason.is_empty() {
+        let items: Vec<Value> = (0..n)
+            .map(|_| {
+                json!({
+                    "create": {
+                        "status": status_code
+                    }
+                })
+            })
+            .collect();
+        json!({
+            "took": took_ms,
+            "errors": false,
+            "items": items,
+        })
+    } else {
+        json!({
+            "took": took_ms,
+            "errors": true,
+            "items": [
+                { "create": { "status": status_code, "error": { "type": "illegal_argument_exception", "reason": error_reason } } }
+            ]
+        })
+    }
+}
+
+/// Returns the headers for every response of Elasticsearch API.
+pub fn elasticsearch_headers() -> HeaderMap {
+    HeaderMap::from_iter([
```

Comment:
Can we use lazy static to create the header map and clone it every time?

Then we can remove this function, but just clone the global variable instead.